packages/order/src/models/transaction.ts

```diff
@@ -0,0 +1,103 @@
+import { BigNumberRawValue, DAL } from "@medusajs/types"
+import {
+  BigNumber,
+  BigNumberField,
+  createPsqlIndexStatementHelper,
+  generateEntityId,
+} from "@medusajs/utils"
+import {
+  BeforeCreate,
+  Cascade,
+  Entity,
+  ManyToOne,
+  OnInit,
+  OptionalProps,
+  PrimaryKey,
+  Property,
+} from "@mikro-orm/core"
+import Order from "./order"
+
+type OptionalLineItemProps = DAL.EntityDateColumns
+
+const ReferenceIdIndex = createPsqlIndexStatementHelper({
+  tableName: "order_transaction",
+  columns: "reference_id",
+})
+
+const OrderIdIndex = createPsqlIndexStatementHelper({
+  tableName: "order_transaction",
+  columns: "order_id",
+})
+
+const CurrencyCodeIndex = createPsqlIndexStatementHelper({
+  tableName: "order_transaction",
+  columns: "currency_code",
+})
+
+@Entity({ tableName: "order_transaction" })
+export default class Transaction {
+  [OptionalProps]?: OptionalLineItemProps
+
+  @PrimaryKey({ columnType: "text" })
+  id: string
+
+  @Property({ columnType: "text" })
+  @OrderIdIndex.MikroORMIndex()
+  order_id: string
+
+  @ManyToOne({
+    entity: () => Order,
+    fieldName: "order_id",
+    cascade: [Cascade.REMOVE, Cascade.PERSIST],
+  })
+  order: Order
+
+  @Property({ columnType: "numeric" })
+  @BigNumberField()
+  amount: BigNumber | number
+
+  @Property({ columnType: "jsonb" })
+  raw_amount: BigNumberRawValue
+
+  @Property({ columnType: "text" })
+  @CurrencyCodeIndex.MikroORMIndex()
+  currency_code: string
+
+  @Property({
+    columnType: "text",
+    nullable: true,
+  })
+  reference: string | null = null
+
+  @Property({
+    columnType: "text",
+    nullable: true,
+  })
+  @ReferenceIdIndex.MikroORMIndex()
+  reference_id: string | null = null
```

Comment:
What does this column represent? If it is the ID in the external system, `external_id` is typically a better name, as `reference` typically means something else and `reference_id` sounds like the two are related.

Reply:
It is the ID of whatever `reference` is.
I'm ok with renaming it, but I would like to also rename `reference`. any suggestion?

Reply:
We discussed with Carlos, this is not a payment reference, but a reference to the originating module/source.

A name we discussed was `originates_from` and `origin_id` as an alternative.

Reply:
I believe `reference` + `reference_id` is a pattern that we've used elsewhere for a similar purpose, so for consistency, I think we should keep it as is

Reply:
@olivermrbl yeah that's what we discussed with carlos, we were just talking if we were to rename it everywhere, what would a better name be.